canvas/canvas.go

```diff
@@ -15,6 +20,19 @@ func Refresh(obj fyne.CanvasObject) {
 	}
 }
 
+// RecolorSVG takes a fyne.Resource containing SVG content, and returns
+// the raw SVG content, re-colorized to be monochrome with the given color.
+// The content can be assigned to a new fyne.StaticResource with an appropriate name
+// to be used in a widget.Button, canvas.Image, etc.
+//
+// If an error occurs, the returned content will be the content of the input Resource,
+// and a non-nil error is returned.
+//
+// Since: 2.6
+func RecolorSVG(resource fyne.Resource, color color.Color) ([]byte, error) {
```

Comment:
It feels like an impedance mismatch to take in a Resource but return a []byte...

If we are expecting people to create an appropriate name for a resource we probably need to be more specific that it should contain the colour name to avoid cache collisions - can we solve this problem for them?

I think it may need to be []byte in and []byte out because this is used internally to ThemedResource handling, so being in canvas package probably implies the lower level API.

Reply:
I am in favor of []byte in and []byte out vs trying to automatically derive a resource name. I'll switch it to that